packages/core/src/sagas/markActivityForSpeakSaga.js

```diff
@@ -1,39 +1,29 @@
 import {
-  cancel,
-  fork,
   put,
-  take
+  takeEvery
 } from 'redux-saga/effects';
 
 import speakableActivity from './definition/speakableActivity';
 import whileConnected from './effects/whileConnected';
+import whileSpeakIncomingActivity from './effects/whileSpeakIncomingActivity';
 
 import { INCOMING_ACTIVITY } from '../actions/incomingActivity';
-import { START_SPEAKING_ACTIVITY } from '../actions/startSpeakingActivity';
-import { STOP_SPEAKING_ACTIVITY } from '../actions/stopSpeakingActivity';
 import markActivity from '../actions/markActivity';
 
 export default function* () {
   yield whileConnected(function* (_, userID) {
-    for (;;) {
-      yield take(START_SPEAKING_ACTIVITY);
-
-      const task = yield fork(markActivityForSpeakSaga, userID);
-
-      yield take(STOP_SPEAKING_ACTIVITY);
-      yield cancel(task);
-    }
+    yield whileSpeakIncomingActivity(markActivityForSpeakSaga.bind(null, userID));
   });
 }
 
 function* markActivityForSpeakSaga(userID) {
-  for (;;) {
-    const { payload: { activity } } = yield take(
-      ({ payload: { activity } = {}, type }) =>
-        type === INCOMING_ACTIVITY
-        && speakableActivity(activity, userID)
-    );
-
-    yield put(markActivity(activity, 'speak', true));
-  }
+  yield takeEvery(
+    ({ payload, type }) => (
```

Comment:
inlining this takes away from the readability for me. can it be defined outside the function and given a good name?

Reply:
Refactored out into a separate function.